frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -335,48 +335,62 @@ sendConfig model initData = Workflow $ do
+                  attrs = ffor useEntireBalance $ \u -> "disabled" =: ("disabled" <$ u)
+              (_, inputAmount, _) <- fmap projectAmount $ uiGasPriceInputField $ cfg
+                & inputElementConfig_elementConfig . elementConfig_modifyAttributes .~ updated attrs
+              useEntireBalance <- case balance of
+                Nothing -> pure $ pure Nothing
+                Just b -> fmap (\v -> b <$ guard v) . _checkbox_value <$> uiCheckbox "input-max-toggle" False def (text "Max")
+              pure (isJust <$> useEntireBalance, liftA2 (<|>) useEntireBalance inputAmount)
 
-          displayImmediateFeedback (updated amount) insufficientFundsMsg $ \ma ->
-            case (ma, _account_status fromAcc) of
-              (Just a, AccountStatus_Exists d) -> a > unAccountBalance (_accountDetails_balance d)
-              (_, _) -> False
+          (useEntireBalance, amount) <- mkLabeledInput True "Amount" gasInputWithMaxButton
+            (def & inputElementConfig_initialValue .~ (maybe "" tshow mInitAmount))
 
-          pure $ runExceptT $ do
-            r <- ExceptT $ first (\_ -> "Invalid Tx Builder") <$> decoded
-            a <- ExceptT $ maybe (Left "Invalid amount") Right <$> amount
+          let insufficientFunds = ffor amount $ \ma -> case (ma, balance) of
+                (Just a, Just b) -> a > b
+                _ -> False
+          displayImmediateFeedback (updated insufficientFunds) insufficientFundsMsg id
 
-            when (maybe True (a >) $ fromAcc ^? account_status . _AccountStatus_Exists . accountDetails_balance . _AccountBalance) $
-              throwError insufficientFundsMsg
+          let validatedTxBuilder = runExceptT $ do
+                r <- ExceptT $ first (\_ -> "Invalid Tx Builder") <$> decoded
+                when (r == TxBuilder fromName fromChain Nothing) $
+                  throwError cannotBeReceiverMsg
+                pure r
 
-            when (r == TxBuilder fromName fromChain Nothing) $
-              throwError cannotBeReceiverMsg
+              validatedAmount = runExceptT $ do
+                a <- ExceptT $ maybe (Left "Invalid amount") Right <$> amount
+                when (maybe True (a >) $ fromAcc ^? account_status . _AccountStatus_Exists . accountDetails_balance . _AccountBalance) $
+                  throwError insufficientFundsMsg
+                pure a
 
-            pure (r, a)
+          pure (validatedTxBuilder, validatedAmount, useEntireBalance)
 
         dialogSectionHeading mempty  "Transaction Settings"
-        (conf, _, _) <- divClass "group" $ uiMetaData model Nothing Nothing
-        pure (conf, recipient)
+        (conf, _, _, _) <- divClass "group" $ uiMetaData model Nothing Nothing
+        pure (conf, useEntireBalance, txBuilder, amount)
       mCaps <- tabPane mempty currentTab SendModalTab_Sign $ do
-        dedrecipient <- eitherDyn $ (fmap . fmap) fst recipient
+        dedrecipient <- eitherDyn txBuilder
```

Comment:
Is it worth putting feedback on this tab that the gas payer can't be the sender because max is clicked?